### DISCUSSION TITLE
How does IINA deal with sample rate?

### DISCUSSION THREAD

**Discussion by Wrestor** [iina/iina#4821] [Q&A] [answered]
Does IINA support automatic sample rate switching, or does it simply match the frequency set in audio-MIDI-Setup? AFAIK macOS will still up/down-sample content that is differing in sample rate if the values set there do not match the source (I've read that Apple Music enthusiasts hate that and use [LosslessSwitcher](https://github.com/vincentneo/LosslessSwitcher) to avoid said behavior on macOS).

**Comment by low-batt** [comment, marked as answer]
For audio/video support IINA uses a library from the [mpv](https://github.com/mpv-player/mpv) project. IINA has some involvement with video, but audio is fully controlled by `mpv`.

Best to ask this question in the [mpv](https://github.com/mpv-player/mpv/issues) project as that is where the developers that work on audio are to be found.

**Reply by krackers** [reply]
@Wrestor I think it will if you set `--coreaudio-change-physical-format` flag.
